### PR TITLE
Pull Request for the Math magicians: events Project

### PR DESCRIPTION
**Pull request by jose-Abel**
Hello,

This is my PR for the Math magicians, events project
- Copied the files calculate.js and operate.js into a logic/ directory in my project.
- Import the files in my Calculator component.
- Implemented the event handlers using the math logic from calculate.js and operate.js in my Calculator React component.
- Used state and props, state in the Calculator.js class base component, props in the Button functional base component.

Cheers.

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+/**
+ * Given a button name and a calculator data object, return an updated
+ * calculator data object.
+ *
+ * Calculator data object contains:
+ *   total:s      the running total
+ *   next:String       the next number to be operated on with the total
+ *   operation:String  +, -, etc.
+ */
```
Comment (barackm) [inline src/logic/calculate.js]: - [x] Can you please remove the comments to keep the code clean 😉

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+  // Button must be an operation
+
+  // When the user presses an operation button without having entered
+  // a number first, do nothing.
+  // if (!obj.next && !obj.total) {
+  //   return {};
+  // }
+
+  // User pressed an operation after pressing '='
```
Comment (barackm) [inline src/logic/calculate.js]: - [x] Can you please remove the comments to keep the code clean 😉

**Review by barackm — CHANGES_REQUESTED** [review]
Hi @jose-Abel, 👋 

You have done an amazing job so far! 👏👏👏
However, there are some issues that you still need to work on to go to the next project. But remember you are almost there 😉  😉 !

## To Highlight
✔️  The code is clean
✔️  Good structure of folders and files
✔️  The calculator logic is working well 👏 

# STATUS: Required Changes ♻️
### _Check the comments under the review._ 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by vichuge — APPROVED** [review]
# Approved project! ✅ 🥳 🎈 🎉🎉🎉🎉🎉🎉🎉🎉 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

InbornCandidHornet-size_restricted

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**